### PR TITLE
feat: Rename fast field precision parameter to fast_precision [Closes #3946]

### LINKED ISSUE #3946: Rename fast field `precision` parameter to `fast_precision`
There is some confusion between the stored precision, which is always in nanoseconds, vs. the fast field precision which is configurable. The current naming scheme does not help.

Let's rename the fast field `precision` parameter to `fast_precision`. We can do so in a backward-compatible manner. The fix should also update the docs and the configuration examples in `config/` and other places (`rg` for `precision: sec|mil|mic|nan`).

See discussion in #3940.

### PR DESCRIPTION
**Pull request by Harsh1s**
Closes #3946 

I have renamed the fast field `precision` parameter to `fast_precision` in a backward-compatible manner. For preserving backward compatibility, I added a `serde alias` attribute on the renamed `fast_precision` field. I also created tests to ensure that the field is deserialized with either `fast_precision` or `precision` as field names in the JSON.

### CODE REVIEW

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs [hunk 1]
```diff
@@ -135,6 +138,47 @@ mod tests {
...
+        )
+        .unwrap();
+
+        assert_eq!(field_mapping_entry.name, "updated_at");
+
+        let date_time_options = match field_mapping_entry.mapping_type {
+            FieldMappingType::DateTime(date_time_options, Cardinality::SingleValue) => {
+                date_time_options
+            }
+            _ => panic!("Expected a date time field mapping."),
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs:165]:
```suggestion
            _ => panic!("expected a date time field mapping"),
```

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs [hunk 2]
```diff
@@ -135,6 +138,47 @@ mod tests {
...
+            fast_precision: DateTimePrecision::Milliseconds,
+            indexed: true,
+            fast: true,
+            stored: false,
+        };
+        assert_eq!(date_time_options, expected_date_time_options);
+    }
+
+    #[test]
+    fn test_date_time_options_single_value_deser_backward_compatible() {
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs:181]: All this backward compatibility test needs to do is deserialize the datetime option and check that the `fast_precison` field was populated correctly. We don't need the remaining duplicate code/test.

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs [hunk 3]
```diff
@@ -176,6 +220,48 @@ mod tests {
...
+            fast_precision: DateTimePrecision::Milliseconds,
+            indexed: true,
+            fast: true,
+            stored: false,
+        };
+        assert_eq!(date_time_options, expected_date_time_options);
+    }
+
+    #[test]
+    fn test_date_time_options_multi_values_deser_backward_compatible() {
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs:264]: We don't need the two additional backward compatibility tests.

**Review by guilload — CHANGES_REQUESTED** [review]
Almost there.

You can run `make fmt` to fix the formatting issues.
You can run `make fix` to fix some of the linting issues.
You can run `make fmt` to test the entire project.

You can run `cargo test --manifest-path quickwit/Cargo.toml -p quickwit-doc-mapper` to test a single crate.

-> [Contributing guidelines](https://github.com/quickwit-oss/quickwit/blob/main/CONTRIBUTING.md)

**Comment by Harsh1s** [comment]
Thanks for your guidance. I have done the changes that you instructed, although I'd like to apologise in advance if didn't do it properly yet again. I have been learning a lot and really am grateful for your patience.

**Review by guilload — APPROVED** [review]

**Comment by guilload** [comment]
Merged, @Harsh1s, thank you!

**Pull request merged**